src/ape_ethereum/provider.py

```diff
@@ -1150,7 +1150,17 @@ def _handle_execution_reverted(
                 **params,
             )
         )
-        return self.compiler_manager.enrich_error(result)
+        enriched = self.compiler_manager.enrich_error(result)
+
+        # Show call trace if availble
+        if enriched.txn:
+            # Unlikely scenario where a transaction is on the error even though a receipt exists.
+            if isinstance(enriched.txn, TransactionAPI) and enriched.txn.receipt:
```

Comment:
we may also want to handle the scenario where a txn exists but a receipt does not, such as during gas estimation

Reply:
You said this in the ape-hardhat PR as well and I guess I'm just not picking up what you're putting down.  What are you suggesting I do in that case?

Reply:
Well, I was going to write down how to do it, but as I began to write down, I realize it is just easier to wait for 0.8 and [the refactor that is happening there](https://github.com/ApeWorX/ape/pull/1864).

So it is not a blocker - but this may not alleviate some of users problems if they are estimating gas and it seems like they are.

But here is some info in case it helps come up with something:

1. The `.trace` may be available already on the txn. In fact, this should be the case if the provider supports `debug_traceCall` as of [this PR](https://github.com/ApeWorX/ape/pull/1951).
2. If the trace is not available on the error for some annoying reason, the same strategy can be applied in [the PR I just mentioned](https://github.com/ApeWorX/ape/pull/1951) to create it (using `debug_traceCall`).
3. Once the trace is procured, a call tree can be created from it using evm-trace lib and stuff, a think a provider does this somewhere as a last resort. But - this is the part that is especially going to be easier with the refactor and which is why I realized it may be ok holding off.
4. Once you have a call tree, you can do the same thing the Receipt is doing to `show` it, if you will.

And again, I realize this process of a bit whonky and the refactor is going to help; I get also that excuse is probably getting hold which is why I have been working on 0.8 a bunch (getting there!).

Reply:
I'm going to just grab the low hanging fruit on this one since you're changing all this stuff anyway.  Let me know if you want me to follow up after/for 0.8.